### PR TITLE
Added unit test for postgres

### PR DESCRIPTION
**Pull request by dislbenn**
Signed-off-by: dislbenn <dbennett@redhat.com>

**Related Issue:**  stolostron/backlog#21297

### Description of changes
Added unit test for postgres deployment/configmap.

### CODE REVIEW

File: controllers/search_test.go [hunk 1]
```diff
@@ -22,6 +23,15 @@ import (
...
 )
 
+func containVolumes(volumes []corev1.Volume, expectedVal string) error {
+	for _, v := range volumes {
+		if v.Name == expectedVal {
+			return nil
+		}
+	}
+	return fmt.Errorf("%s does not exist within the listed volumes env", expectedVal)
+}
```
Comment (jlpadilla) [inline controllers/search_test.go:33]: You could pass `t` and validate within this function.

```suggestion
func verifyContainsVolume(t *testing.T, volumes []corev1.Volume, expectedVolume string) {
	for _, v := range volumes {
		if v.Name == expectedVolume {
			return
		}
	}
	t.Errorf("Failed to find to find volume: %s", expectedVolume)
}
```

File: controllers/search_test.go [hunk 2]
```diff
@@ -118,6 +140,16 @@ func TestSearch_controller(t *testing.T) {
...
+	//check for configmap
+	configmap3 := &corev1.ConfigMap{}
+	err = cl.Get(context.TODO(), types.NamespacedName{
+		Name: "search-postgres",
+	}, configmap3)
+
+	if err != nil {
+		t.Fatalf("Failed to get configmap %s: %v", "search-postgres", err)
+	}
+
```
Comment (jlpadilla) [inline controllers/search_test.go:152]: Could we also verify the contents of the configMap?

Comment (dislbenn) [reply controllers/search_test.go:157]: Updated the test to verify certain contents of the configMap.

File: controllers/search_test.go (hunk 1 above)
Comment (dislbenn) [reply controllers/search_test.go]: Updated the test function to handle the validation.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

**Pull request merged**